axelrod/strategies/dbs.py

```diff
@@ -0,0 +1,390 @@
+from axelrod.actions import Actions, Action
+from axelrod.player import Player
+
+
+C, D = Actions.C, Actions.D
+
+
+class DBS(Player):
+    """
+    Desired Belief Strategy as described in [Au2006]_
+    http://www.cs.utexas.edu/%7Echiu/papers/Au06NoisyIPD.pdf
+
+    A strategy that learns the opponent's strategy, and uses symbolic 
+    noise detection for detecting whether anomalies in player’s behavior
+    are deliberate or accidental, hence increasing performance in noisy 
+    tournaments.  
+
+    From the learned opponent's strategy, a tree search is used to
+    choose the best move
+
+    Default values for the parameters are the suggested values in the
+    article. When noise increases you can try to diminish 
+    violation_threshold and rejection_threshold
+
+    Parameters
+    ----------
+    discount_factor : float, optional
+        used when computing discounted frequencies to learn opponent's
+        strategy. Must be between 0 and 1. The default is 0.75
+    promotion_threshold : int, optional 
+        number of observations needed to promote a change in opponent's
+        strategy. The default is 3.
+    violation_threshold : int, optional
+        number of observations needed to considerate opponent's 
+        strategy has changed. You can lower it when noise increases.
+        The default is 4, which is good for a noise level of .1 
+    reject_threshold : int, optional
+        number of observations before forgetting opponent's previous
+        strategy. You can lower it when noise increases. The default 
+        is 3, which is good for a noise level of .1
+    tree_depth: int, optional
+        depth of the tree for the tree-search algorithm. Higher depth
+        means more time to compute the move. The default is 5
+    """
+
+    # These are various properties for the strategy
+    name = 'DBS'
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self, discount_factor=.75, promotion_threshold=3, 
+                 violation_threshold=4, reject_threshold=3, tree_depth=5): 
+        super().__init__()
+
+        # default opponent's policy is TitForTat
+        self.Rd = create_policy(1, 1, 0, 0)
```

Comment:
We've called policies "four vectors" elsewhere (in strategies/memoryone.py), are they always dictionaries of size 4?